tests/verify/test_verifier_verify.py

```diff
@@ -433,3 +434,241 @@ def test_verify(
 ):
     mock_exists(True)
     assert verifier.verify() == VerifyCommandResult.model_validate(expected)
+
+
+class TimeoutPerfCounter:
+    """Mock perf_counter that returns values from a sequence for timeout testing"""
+
+    def __init__(self, sequence: list[float]) -> None:
+        self.sequence = sequence
+        self.call_count = 0
+
+    def __call__(self) -> float:
+        if self.call_count < len(self.sequence):
+            result = self.sequence[self.call_count]
+            self.call_count += 1
+            return result
+        return self.sequence[-1] + (self.call_count - len(self.sequence) + 1)
```

Comment:
This class is likely unnecessary. The same functionality should be achievable by assigning a list to `side_effect` on the mock.

https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.side_effect